presto-main/src/main/java/io/prestosql/operator/WindowOperator.java

```diff
@@ -303,6 +312,13 @@ public WindowOperator(
                     .flatMap(this::pagesIndexToWindowPartitions)
                     .transform(new WindowPartitionsToOutputPages());
         }
+        else if (limitWindowFunction.isPresent()) {
```

Comment:
With this condition, it looks like if the disk spill is enabled, LimitWindowPartitionToOutputPages will never be used.

Reply:
it’s an intentional as we would not use disk spill in the near futur